src/NUnitFramework/framework/Constraints/EqualStringWithoutUsingConstraint.cs

```diff
@@ -123,18 +125,63 @@ public sealed override ConstraintResult ApplyTo<TActual>(TActual actual)
             {
                 hasSucceeded = _expected is null;
             }
+            else if (actual is string actualString)
+            {
+                return ApplyTo(actualString);
+            }
+            else if (CanCastToString(actual, out string? actualCastToString))
+            {
+                return ApplyTo(actualCastToString);
+            }
+            else if (actual is IEquatable<string> equatableString)
+            {
+                if (_caseInsensitive || _ignoringWhiteSpace)
+                {
+                    throw new InvalidOperationException("Cannot use IgnoreCase or IgnoreWhiteSpace with IEquatable<string>.");
+                }
+
+                hasSucceeded = equatableString.Equals(_expected);
+            }
             else if (_expected is null)
             {
                 hasSucceeded = false;
             }
             else
             {
-                return ApplyTo(actual as string);
+#if TRUE
+                hasSucceeded = false;
+#else
+                // Alternatively we could fall back to pre 4.3 EqualConstraint behavior
+                // But if the actual value cannot be convert to a string nor can be compared to one
+                // we should fail the test.
+                return new EqualConstraint(_expected).ApplyTo(actual);
+#endif
             }
 
             return ConstraintResult(actual, hasSucceeded);
         }
 
+        private static bool CanCastToString<TActual>(TActual actual, out string? s)
+        {
+            // Check if the type implements an implicit cast to string
+            // Note that we don't have to check the parameter types
+            // as the compiler only allows cast from/to TActual and we check the return type.
+            var implicitCastToString = typeof(TActual).GetMethods()
+                                                      .Where(x => x.IsStatic &&
+                                                                  x.Name == "op_Implicit" &&
+                                                                  x.ReturnType == typeof(string))
+                                                      .SingleOrDefault();
```

Comment:
The problem with `FirstOrDefault` might pick the wrong one and throw an exception in the `Invoke` due to an argument mismatch.

The [C# language spec](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/classes#15104-conversion-operators) says it cannot be done:

> From the second rule, it follows that a conversion operator shall convert either to or from the class or struct type in which the operator is declared.
>>It is possible for a class or struct type C to define a conversion from C to int and from int to C, but not from int to bool.

So if the return type is `string` then according to the specification, the argument must be the class at hand.
Hence I prefer to keep it until the spec changes here.

Reply:
Thanks @manfred-brands that makes sense to me. I agree it's safe.

Reply:
> I've tried to "break" this and haven't been able to, so I think the `SingleOrDefault()` is fine here. Still, I don't know if we'd prefer to use `FirstOrDefault()` just in case this can ever find two methods?

Suggesting a small performance improvement:
As you mentioned, in C# a class cannot have multiple implicit operators with the same return type. The compiler enforces uniqueness. This means there will be at most one match for `op_Implicit` returning a `string`.

`FirstOrDefault` stops searching as soon as the first match is found, avoiding unnecessary scanning of the remaining methods - as `SingleOrDefault` does to enforce uniqueness.

So consider using this instead:
```csharp
typeof(TActual).GetMethods()
    .FirstOrDefault(x => x.IsStatic &&
        x.Name == "op_Implicit" &&
        x.ReturnType == typeof(string));
```

Reply:
Thanks @moshekar, change made. With RedisValue having a dozen of those it might help.

Reply:
Thanks @manfred-brands .